elytron/src/main/java/org/wildfly/extension/elytron/AggregateRealmDefinition.java

```diff
@@ -122,18 +131,39 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
             final List<InjectedValue<SecurityRealm>> authorizationRealmValues = new ArrayList<>();
             ModelNode authorizationRealmNode = AUTHORIZATION_REALM.resolveModelAttribute(context, model);
 
+            String principalTransformer = PRINCIPAL_TRANSFORMER.resolveModelAttribute(context, model).asStringOrNull();
+
+            InjectedValue<PrincipalTransformer> principalTransformerValue = null;
+            String principalTransformerRuntimeCapability;
+            ServiceName principalTransformerServiceName = null;
+
+            if (principalTransformer != null) {
+                principalTransformerValue = new InjectedValue<PrincipalTransformer>();
+                principalTransformerRuntimeCapability = RuntimeCapability.buildDynamicCapabilityName(PRINCIPAL_TRANSFORMER_CAPABILITY, principalTransformer);
+                principalTransformerServiceName = context.getCapabilityServiceName(principalTransformerRuntimeCapability, PrincipalTransformer.class);
+            }
+
+            InjectedValue<PrincipalTransformer> finalPrincipalTransformerValue = principalTransformerValue;
             TrivialService<SecurityRealm> aggregateRealmService = new TrivialService<SecurityRealm>(() -> {
                 SecurityRealm[] authorizationRealms = new SecurityRealm[authorizationRealmValues.size()];
                 for (int i = 0; i < authorizationRealms.length; i++) {
                     authorizationRealms[i] = authorizationRealmValues.get(i).getValue();
                 }
 
-                return new AggregateSecurityRealm(authenticationRealmValue.getValue(), authorizationRealms);
+                if (principalTransformer != null) {
```

Comment:
This null check should be done on ```principalTransformerValue``` since the next line is going to call ```principalTransformerValue.getValue()```.

Reply:
fixed